ds13/code/modules/mob/living/psychosis.dm

```diff
@@ -0,0 +1,182 @@
+	var/psychosis_immune = FALSE //SET THIS TO TRUE FOR NECROMORPHS IF YOU MAKE THEM A HUMAN SUBTYPE, OR THEY WILL GO INSANE FROM MARKER STUFF
+	var/sanity_cooldown = FALSE
+
+/mob/living/carbon/human/Life()
+	. = ..() //Override here to allow mobs to slowly heal from psychosis damage...SLOWLY.
+	if(sanity < 100)
+		adjustPsychosisLoss(-0.25) //From 0 sanity to 100 on this alone will take about 10 minutes. While you can't drug the symptoms away, you can remove the hallucinations with Synaptizine.
+
+/mob/living/carbon/human/proc/getPsychosisLoss()
+	return sanity
+
+/mob/living/carbon/human/proc/adjustPsychosisLoss(var/amount)
+	if (status_flags & GODMODE || psychosis_immune)
+		return FALSE
+	if(stat != CONSCIOUS) //Let's be realistic here.
+		return FALSE
+	sanity = Clamp(sanity - amount, 0, maxSanity)
+	if(sanity < last_sanity) //If they're on the mend, there's no need to give them the effects of a psychotic episode. This prevents their healing from psychosis damage from actually harming them with hallucinations.
+		switch(last_sanity)
+			if(0 to 29) //You will get fucked up at this point.
+				set_psychosis(3)
+			if(30 to 59) //You're definitely not feeling OK.
+				set_psychosis(2)
+			if(60 to 80) //Feeling slightly off
+				set_psychosis(1)
+			if(81 to 100)
+				set_psychosis(0)
+	else if(sanity >= 90) //You're back to normal, and the bad times have worn off.
+		set_psychosis(0)
+	last_sanity = sanity
+	return TRUE
+
+/mob/living/carbon/human/proc/setPsychosisLoss(var/amount)
+	adjustPsychosisLoss((amount * 0.5)-getPsychosisLoss())
+
+//Ghetto components until we get them here. This does the job, but isn't the most efficient solution.//
+
+
+/**
+*
+*
+*
+* Trauma pseudo-components. These attach to an object and will process VERY slowly, applying psychosis damage to anyone who's unlucky enough to see them. This does mean you can
+* blindfold yourself to avoid going insane.
+*
+*
+*
+*
+*/
+
+/atom/proc/set_traumatic_sight(var/isTrauma)
+	if(isTrauma)
+		SStrauma.try_add_trauma(src)
+	else
+		SStrauma.try_remove_trauma(src)
+
+/datum/component //Placeholder for when we get components.
+
+#define TRAUMA_POWER_WEAK 1.50
+#define TRAUMA_POWER_MEDIUM 3
+#define TRAUMA_POWER_STRONG 5
+
+/datum/component/traumatic_sight
+	var/name = "Traumatic sight handler"
+	var/atom/source = null
+	var/trauma_power = 1.50 //How traumatic is this sight? Dealt to them per traumatic object, per 10 seconds. A room full of necros would seriously fuck you up. A "STRONG" trauma would have a value of 5, and would give you just over a minute to cover your eyes before it started to affect you. Better bring that blindfold with you!
+
+/datum/component/traumatic_sight/New()
+	. = ..()
+	START_PROCESSING(SStrauma, src)
+
+/datum/component/traumatic_sight/Process() //This isn't terribly efficient, but it's also not the worst performance wise.
+	for(var/mob/living/carbon/human/H in viewers(source, null))
```

Comment:
did you use step triggers like i advised? Having these things polling regularly is not good.
I realise its a slow polling, but that also means things are very unresponsive.